planning/behavior_path_planner/src/planner_manager.cpp

```diff
@@ -41,6 +42,46 @@ PlannerManager::PlannerManager(
   debug_publisher_ptr_ = std::make_unique<DebugPublisher>(&node, "~/debug");
 }
 
+void PlannerManager::launchScenePlugin(rclcpp::Node & node, const std::string & name)
+{
+  if (plugin_loader_.isClassAvailable(name)) {
+    const auto plugin = plugin_loader_.createSharedInstance(name);
+    plugin->init(&node);
+
+    // Check if the plugin is already registered.
+    for (const auto & running_plugin : manager_ptrs_) {
+      if (plugin->name() == running_plugin->name()) {
+        RCLCPP_WARN_STREAM(node.get_logger(), "The plugin '" << name << "' is already loaded.");
+        return;
+      }
+    }
+
+    // register
+    manager_ptrs_.push_back(plugin);
+    processing_time_.emplace(plugin->name(), 0.0);
+    RCLCPP_INFO_STREAM(node.get_logger(), "The scene plugin '" << name << "' is loaded.");
+  } else {
+    RCLCPP_ERROR_STREAM(node.get_logger(), "The scene plugin '" << name << "' is not available.");
+  }
+}
+
+void PlannerManager::removeScenePlugin(rclcpp::Node & node, const std::string & name)
+{
+  auto it = std::remove_if(manager_ptrs_.begin(), manager_ptrs_.end(), [&](const auto plugin) {
+    return plugin->name() == name;
+  });
+
+  if (it == manager_ptrs_.end()) {
+    RCLCPP_WARN_STREAM(
+      node.get_logger(),
+      "The scene plugin '" << name << "' is not found in the registered modules.");
+  } else {
+    manager_ptrs_.erase(it, manager_ptrs_.end());
+    processing_time_.erase(name);
+    RCLCPP_INFO_STREAM(node.get_logger(), "The scene plugin '" << name << "' is unloaded.");
+  }
+}
+
```

Comment:
just a question. 
this is not currently used, but is this a mechanism to dynamically disable modules in the future?

Reply:
YES.